dotnet/src/SemanticKernel/AI/OpenAI/Clients/OpenAIClientAbstract.cs

```diff
@@ -132,14 +143,18 @@ protected virtual void Dispose(bool disposing)
     // HTTP user agent sent to remote endpoints
     private const string HTTPUseragent = "Microsoft Semantic Kernel";
 
-    private async Task<T> ExecutePostRequestAsync<T>(string url, string requestBody)
+    private async Task<T> ExecutePostRequestAsync<T>(string url, string requestBody, int httpTimeoutInSeconds)
     {
+        if (httpTimeoutInSeconds < 1) throw new AIException(AIException.ErrorCodes.UnknownError, "Invalid http timeout");
```

Comment:
Might be worth defining a new ErrorCode if there isn't a relevant one already (InvalidRequest, maybe?)